### PR TITLE
feat: Officially support web

### LINKED ISSUE #58: Disable browser session tracking
As this SDK wraps the browser SDK, the auto session tracking is running on both the native layer and the JS layer. Although nothing is sent by the browser SDK because of:
```
[warn] - Sentry Logger [Warn]: Dropping session because custom transport doesn't implement sendSession
```
we should disable the browser session tracking.

### LINKED ISSUE #63: Support Web
### Environment

How do you use Sentry?
Sentry SaaS

Which SDK and version?
@sentry/capacitor@0.3.0

### Steps to Reproduce

1. Followed the README
2. Started my Ionic app in the browser

### Expected Result

Fallback to JS SDK (Angular is my case)?

### Actual Result

```
instrument.js:109 Unhandled Promise rejection: Native Client is not available, can't start on native. ; Zone: <root> ; Task: null ; Value: SentryError: Native Client is not available, can't start on native.
    at new SentryError (error.js:8)
    at Object.35046 (wrapper.js:291)
    at __webpack_require__ (bootstrap:19)
    at Object.51297 (sdkinfo.js:6)
    at __webpack_require__ (bootstrap:19)
    at Object.56889 (index.js:5)
    at __webpack_require__ (bootstrap:19)
    at Object.39167 (sdk.js:7)
    at __webpack_require__ (bootstrap:19)
    at Object.79555 (index.js:29) SentryError: Native Client is not available, can't start on native.
    at new SentryError (http://localhost:8100/vendor.js:90528:28)
    at Object.35046 (http://localhost:8100/vendor.js:85627:25)
    at __webpack_require__ (http://localhost:8100/runtime.js:23:42)
    at Object.51297 (http://localhost:8100/vendor.js:85023:17)
    at __webpack_require__ (http://localhost:8100/runtime.js:23:42)
    at Object.56889 (http://localhost:8100/vendor.js:85006:17)
    at __webpack_require__ (http://localhost:8100/runtime.js:23:42)
    at Object.39167 (http://localhost:8100/vendor.js:85210:22)
    at __webpack_require__ (http://localhost:8100/runtime.js:23:42)
    at Object.79555 (http://localhost:8100/vendor.js:84950:13)
(anonymous) @ instrument.js:109
api.onUnhandledError @ zone.js:1063
handleUnhandledRejection @ zone.js:1088
api.microtaskDrainDone @ zone.js:1082
drainMicroTaskQueue @ zone.js:589
Promise.then (async)
scheduleMicroTask @ zone.js:565
scheduleTask @ zone.js:396
scheduleTask @ zone.js:221
scheduleMicroTask @ zone.js:241
scheduleResolveOrReject @ zone.js:1266
then @ zone.js:1410
setDriver @ localforage.js:2734
LocalForage @ localforage.js:2523
4.3 @ localforage.js:2796
s @ localforage.js:7
e @ localforage.js:7
(anonymous) @ localforage.js:7
(anonymous) @ localforage.js:7
92430 @ localforage.js:7
__webpack_require__ @ bootstrap:19
21014 @ vendor.js:88971
__webpack_require__ @ bootstrap:19
39462 @ vendor.js:88934
__webpack_require__ @ bootstrap:19
39167 @ sdk.js:6
__webpack_require__ @ bootstrap:19
79555 @ index.js:29
__webpack_require__ @ bootstrap:19
14431 @ environment.ts:21
__webpack_require__ @ bootstrap:19
__webpack_exec__ @ main.js:4245
(anonymous) @ main.js:4246
__webpack_require__.O @ chunk loaded:25
(anonymous) @ main.js:4247
webpackJsonpCallback @ jsonp chunk loading:72
(anonymous) @ main.js:1
error.js:8 Uncaught SentryError: Native Client is not available, can't start on native.
    at new SentryError (error.js:8)
    at Object.35046 (wrapper.js:291)
    at __webpack_require__ (bootstrap:19)
    at Object.51297 (sdkinfo.js:6)
    at __webpack_require__ (bootstrap:19)
    at Object.56889 (index.js:5)
    at __webpack_require__ (bootstrap:19)
    at Object.39167 (sdk.js:7)
    at __webpack_require__ (bootstrap:19)
    at Object.79555 (index.js:29)
SentryError @ error.js:8
35046 @ wrapper.js:291
__webpack_require__ @ bootstrap:19
51297 @ sdkinfo.js:6
__webpack_require__ @ bootstrap:19
56889 @ index.js:5
__webpack_require__ @ bootstrap:19
39167 @ sdk.js:7
__webpack_require__ @ bootstrap:19
79555 @ index.js:29
__webpack_require__ @ bootstrap:19
14431 @ environment.ts:21
__webpack_require__ @ bootstrap:19
__webpack_exec__ @ main.js:4245
(anonymous) @ main.js:4246
__webpack_require__.O @ chunk loaded:25
(anonymous) @ main.js:4247
webpackJsonpCallback @ jsonp chunk loading:72
(anonymous) @ main.js:1
zone.js:372 Uncaught (in promise) SentryError: Native Client is not available, can't start on native.
    at new SentryError (error.js:8)
    at Object.35046 (wrapper.js:291)
    at __webpack_require__ (bootstrap:19)
    at Object.51297 (sdkinfo.js:6)
    at __webpack_require__ (bootstrap:19)
    at Object.56889 (index.js:5)
    at __webpack_require__ (bootstrap:19)
    at Object.39167 (sdk.js:7)
    at __webpack_require__ (bootstrap:19)
    at Object.79555 (index.js:29)
```

### PR DESCRIPTION
**Pull request by jennmueng**
Add official support for the web platform. Also disables the native session tracking and relies on browser SDK's session tracking when running on web and vice versa.

Resolves #63
Fixes #58 

Added unit tests to verify.

### CODE REVIEW

File: src/sdk.ts [hunk 1]
```diff
@@ -12,10 +12,22 @@ import { CapacitorScope } from './scope';
...
 import { NATIVE } from './wrapper';
 
-const DEFAULT_OPTIONS: CapacitorOptions = {
+export const DEFAULT_OPTIONS_WEB: CapacitorOptions = {
+  enableNativeNagger: false,
+  enableNative: false,
+
+  // Use the browser's session tracking instrumentation on web.
+  enableAutoSessionTracking: false,
+  autoSessionTracking: true,
```
Comment (AbhiPrasad) [inline src/sdk.ts:21]: Can we rename these somehow? It's weird that they feel like the same exact option right now.

File: test/sdk.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  it('Uses default mobile options on android', () => {
+    NATIVE.platform = 'android';
+
+    init({ dsn: '' }, (finalOptions: CapacitorOptions) => {
+      expect(finalOptions).toEqual(
+        expect.objectContaining(DEFAULT_OPTIONS_MOBILE),
+      );
+    });
+  });
+});
```
Comment (AbhiPrasad) [inline test/sdk.test.ts:45]: Let's table test this 👀 

```ts
// [string, platform, options]
const table: Array<[string, string, CapacitorOptions]> = [
  ['Uses default web options on web', 'web', DEFAULT_OPTIONS_WEB],
  ['Uses default mobile options on ios', 'ios', DEFAULT_OPTIONS_MOBILE],
  ['Uses default mobile options on android', 'android', DEFAULT_OPTIONS_MOBILE],
];

it.each(table)('%s', (...test) => {
  NATIVE.platform = test[1];
  init({ dsn: '' }, (finalOptions: CapacitorOptions) => {
    expect(finalOptions.enableNative).toBe(false);
    expect(finalOptions).toEqual(expect.objectContaining(test[2]));
  });
});
```

**Review by AbhiPrasad — CHANGES_REQUESTED** [review]

File: test/sdk.test.ts (hunk 2 above)
Comment (jennmueng) [reply test/sdk.test.ts:45]: Oh I did not know about this, would've been handy in lots of places lol

File: src/sdk.ts (hunk 1 above)
Comment (jennmueng) [reply src/sdk.ts:21]: How about we have it be `enableNativeSessionTracking` and `enableBrowserSessionTracking`? The init call here in cap would just convert the two into the respective options for the browser sdk and the native bridge? cc @bruno-garcia

Comment (AbhiPrasad) [reply src/sdk.ts:21]: I like splitting it up that way. 

Should we even refer to native at all, maybe just call it `enableMobileSessionTracking`? I think we should do whatever we think will make it easier for users to understand.

Comment (jennmueng) [reply src/sdk.ts:21]: @AbhiPrasad In that sense, let's just have a single `enableAutoSessionTracking` flag for the users, and the capacitor init just determines to enable native or browser session tracking without needing the user to know about it.

File: test/sdk.test.ts [hunk 3]
```diff
@@ -1,45 +1,81 @@
...
 
-    init({ dsn: '' }, (finalOptions: CapacitorOptions) => {
-      expect(finalOptions.enableNative).toBe(false);
-      expect(finalOptions).toEqual(
-        expect.objectContaining(DEFAULT_OPTIONS_WEB),
-      );
-    });
-  });
+describe('SDK Init', () => {
+  // [string, platform, options, autoSessionTracking, enableNative, enableAutoSessionTracking]
```
Comment (AbhiPrasad) [inline test/sdk.test.ts:21]:
```suggestion
  // [name, platform, options, autoSessionTracking, enableNative, enableAutoSessionTracking]
```

File: src/sdk.ts [hunk 4]
```diff
@@ -86,9 +75,20 @@ export function init<O>(
...
+
+  const mobileOptions = {
+    ...finalOptions,
+    enableAutoSessionTracking:
+      NATIVE.platform !== 'web' && finalOptions.enableAutoSessionTracking,
+  } as CapacitorOptions;
 
-  void NATIVE.initNativeSdk(finalOptions);
+  originalInit(browserOptions);
+  void NATIVE.initNativeSdk(mobileOptions);
```
Comment (AbhiPrasad) [inline src/sdk.ts:91]: Do we call this for web?

File: test/sdk.test.ts [hunk 5]
```diff
@@ -1,45 +1,81 @@
...
+      true,
+      false,
+    ],
+    [
+      'enableAutoSessionTracking sets enableAutoSessionTracking on ios',
+      'ios',
+      { dsn: '', enableAutoSessionTracking: false },
+      false,
+      true,
+      false,
```
Comment (AbhiPrasad) [inline test/sdk.test.ts:62]: Let’s make these last two booleans -> the object containing enableNative and enableAutoSessionTracking

File: test/sdk.test.ts [hunk 6]
```diff
@@ -1,45 +1,81 @@
...
-        expect.objectContaining(DEFAULT_OPTIONS_WEB),
-      );
-    });
-  });
+describe('SDK Init', () => {
+  // [string, platform, options, autoSessionTracking, enableNative, enableAutoSessionTracking]
+  const table: Array<[
+    string,
+    string,
+    CapacitorOptions & BrowserOptions,
```
Comment (AbhiPrasad) [inline test/sdk.test.ts:25]: Shouldn’t this be a union type?

File: src/sdk.ts (hunk 4 above)
Comment (jennmueng) [reply src/sdk.ts:91]: Yep, and if `enableNative = false` it sets `NATIVE.enableNative` to false and then the native wrapper does not do anything.

File: test/sdk.test.ts (hunk 6 above)
Comment (jennmueng) [reply test/sdk.test.ts:25]: I don't think so, it should be this: https://github.com/getsentry/sentry-capacitor/blob/36fd6a585d2f6f391f937e44411c7cfb7c75a10c/src/sdk.ts#L22

Which should be an intersection of `CapacitorOptions` and the options for the sibling SDK

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**